pkg/pillar/cmd/volumemgr/handlevolume.go

```diff
@@ -228,16 +227,17 @@ func deleteVolume(ctx *volumemgrContext, status *types.VolumeStatus) {
 			}
 		} else {
 			log.Infof("VolumeWorkResult(%s) not found", status.Key())
+			// XXX what happens when VolumeWork is done?
 		}
 	}
 	publishVolumeStatus(ctx, status)
 	unpublishVolumeStatus(ctx, status)
-	log.Infof("deleteVolume for %v Done", status.Key())
+	log.Infof("maybeDeleteVolume for %v Done", status.Key())
 }
 
 // updateVolumeStatusRefCount updates the refcount in volume status
 // Refcount in volume status is sum of refount in volume config and volume ref config
-func updateVolumeStatusRefCount(ctx *volumemgrContext, vs *types.VolumeStatus) *types.VolumeStatus {
+func updateVolumeStatusRefCount(ctx *volumemgrContext, vs *types.VolumeStatus) {
```

Comment:
The reason for only updating the refcount and not publishing from here is that there might be a possibility that publishers get stuck in the select process. And, immediately after updating the refcount we call ```deleteVolume``` function. So, in my opinion, it can introduce some bugs here.

Reply:
It is not publishing; it is updating the object.
But returning a pointer to an object passed by reference is just confusing. That only makes sense if sometimes the function would return a different pointer. In this case it doesn't.

Hence this change just removes that odd calling/return approach. 
If we need to track whether or not to publish this function can be made to return a changed boolean to the caller.

But re-reading the callers I see handleVolumeRefDelete is not publishing after calling this function.

Didn't understand your "stuck in select" concern.

Reply:
My bad. I didn't see that. Sorry for the confusion.